src/test/java/com/product/productService/ProductServiceApplicationTest.java

```diff
@@ -0,0 +1,176 @@
+package com.product.productService;
+
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.google.common.collect.ImmutableList;
+import com.product.dao.ProductDaoImpl;
+import com.product.model.Manufacturer;
+import com.product.model.Order;
+import com.product.model.Product;
+import com.product.service.ProductServiceImpl;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductServiceApplicationTest {
+	
+	@Mock
+	private ProductDaoImpl productDaoImpl;
+
+	@InjectMocks
+	private ProductServiceImpl productServiceImpl;
+
+	private List<Product> productList;
+	private  Product product1;
+	private  Product product2;
+
+	@Before
+	public void preRequisiteDataCreation() {
+
+		/* Data Creation */
+		ImmutableList<String> othersize = ImmutableList.of("Rick", "Ron", "Victor");
+		Map<String, Integer> details = new LinkedHashMap<String, Integer>();
```

Comment:
Does it need to maintain the order ??

Reply:
yes.

Reply:
why ? this is details .. who cares ?

I saw u used `mmutableMap<String, Integer> details = ImmutableMap.of("quantity", 1, "sold",2);` in the other test class. Why not in this one ?